main.js

```diff
@@ -0,0 +1,71 @@
+
+I also know that the comment reply form is showing up at the end of the posts div instead of directly after
+the comment that the "Add Comment" button is clicked on. I tried moving it to the end of the post div, but
+without the unique ID number issue fixed and functioning, I couldn't figure out how to append the form
+to the single unique post (which is the same thing that is happening when I do reply to an existing comment,
+it posts my reply to all comments instead of just the single one it is supposed to - I know this is related to
+the unique ID issue) - considered adding the same index on the end of the add-comment button click
+function and iterate through using i and i++, but couldn't figure out the correct way to do the syntax. 
+
+There were alot of challenges for me on this project. Conceptually I feel like I know what I need to do, I 
+just am really struggling with the syntax and specific way to write what needs to happen.
+Will continue to work through these issues but wanted to commit where I was at.
+*/
+
+//When Post button is clicked, this will post the message as a parent thread.
+$('#submit').on('click', function () {
+  //Gets the user input for both the name and message boxes.
+  var $name = $('#name').val();
+  var $text = $('#message').val();
+
+  //Creates a div, then one paragraph element to contain the message, a second paragraph element
+  //to contain the poster name, then adds on a button for Removing the Comment and a buttong for
+  //adding a comment to that specific comment.
+  $('.posts').append('<div class="post"><p>' + $text + '<p>Posted By: ' + $name + '<p><button class="remove-button" id="removeButton">Delete Comment</button>' + '<button class="add-comments-button" id="addCommentsButton">Add Comment</button>' + '</p></p></p></div>');
+
+  //Adds on a blank comment form to the comment but hides it. 
+  $('#blank-comment-form').hide();
+  $('.posts').append($('#blank-comment-form'));
+
+  //Adds a horizontal line after hidden comment box
+  $('.posts').append('<hr>');
+
+   //Resets the form fields back to blank
+   $('form :input').val('');
+});
+
+//When Post Reply button is clicked, this will post the message as reply to the specific comment.
+$('#submit-comment').on('click', function () {
+  //Gets the user input for both the name and message boxes.
+  var $commentName = $('#comment-name').val();
+  var $commentText = $('#comment-message').val();
+
+  //Creates a div, then one paragraph element to contain the message, a second paragraph element
+  //to contain the poster name, then adds on a button for Removing the Comment and a buttong for
+  //adding a comment to that specific comment.
+  $('.post').append('<div class="sub-post"><p>' + $commentText + '<p>Posted By: ' + $commentName + '<p><button class="remove-button" id="removeButton">Delete Comment</button>' + '<button class="add-comments-button" id="addCommentsButton">Add Comment</button>' + '</p></p></p></div>');
```

Comment:
I think the whole comment form thing is what messed you up. I would add a comment form to each post right here, when you append the HTML. You weren't able to get to the rest of this because this blocked you.

In the future, pos in Slack and ask for advice when you're stuck. In this case I think this route could have been suggested to you and you would have been off to the next problem, which is posting comments. Based on your comments above, I think you had a good strategy in mind for that next part.